integration/experiment/gpu_frequency_sweep/gpu_frequency_sweep.py

```diff
@@ -0,0 +1,153 @@
+    parser.add_argument('--step-gpu-frequency', dest='step_gpu_frequency',
+                        action='store', type=float, default=None,
+                        help='increment in hertz between gpu frequency settings for the sweep')
+
+    parser.set_defaults(agent_list='frequency_map')
+
+
+def setup_gpu_frequency_bounds(mach, min_gpu_freq, max_gpu_freq,
+                                  step_gpu_freq):
+    sys_min = mach.gpu_frequency_min()
+    sys_max = mach.gpu_frequency_max()
+    #TODO: Replace with gpu step when signal is merged in
+    sys_step = mach.frequency_step()
+    if min_gpu_freq is None:
+        min_gpu_freq = sys_min
+    if max_gpu_freq is None:
+        max_gpu_freq = sys_max
+    if step_gpu_freq is None:
+        step_gpu_freq = sys_step
+    if min_gpu_freq < sys_min or max_gpu_freq > sys_max:
+        raise RuntimeError('GPU frequency bounds are out of range for this system')
+    if min_gpu_freq > max_gpu_freq:
+        raise RuntimeError('GPU frequency min is greater than max')
+
+    gpu_freqs = list(reversed(experiment_steps(min_gpu_freq, max_gpu_freq, step_gpu_freq)))
+    return gpu_freqs
+
+def experiment_steps(minv, maxv, stepv):
+    rval = [minv]
+    while rval[-1] + stepv <= maxv:
+        rval.append(rval[-1] + stepv)
+    return rval
+
+def report_signals():
+    return ["CPU_CYCLES_THREAD@package", "CPU_CYCLES_REFERENCE@package",
+            "TIME@package", "CPU_ENERGY@package", "GPU_CORE_FREQUENCY_STATUS@board"]
+
+
+def trace_signals():
+    return [ "GPU_CORE_FREQUENCY_STATUS@gpu" ]
+
+def launch_configs(output_dir, app_conf, core_freq_range, uncore_freq_range, gpu_freq_range):
+    agent = 'frequency_map'
+    targets = []
+    for freq in core_freq_range:
+        for uncore_freq in uncore_freq_range:
+            for gpu_freq in gpu_freq_range:
+                name = f'{freq:.1e}c_{uncore_freq:.1e}u_{gpu_freq:.1e}g'
+                options = {'FREQ_CPU_DEFAULT': freq,
+                           'FREQ_UNCORE': uncore_freq,
+                           'FREQ_GPU_DEFAULT': gpu_freq}
+                file_name = os.path.join(output_dir, f'{agent}_agent_{name}.config')
+                agent_conf = geopmpy.agent.AgentConf(file_name, agent, options)
+                targets.append(launch_util.LaunchConfig(app_conf=app_conf,
+                                                    agent_conf=agent_conf,
+                                                    name=name))
+    return targets
+
+def launch(app_conf, args, experiment_cli_args):
+    '''
+    Run the application over a range of core, uncore, and gpu frequencies.
+    '''
+
+    print(f"Core freq: {args.min_frequency} - {args.max_frequency} step {args.step_frequency},\n"
+          f"Uncore: {args.min_uncore_frequency} - {args.max_uncore_frequency}, step {args.step_uncore_frequency}.\n"
+          f"GPU: {args.min_gpu_frequency} - {args.max_gpu_frequency}, step {args.step_gpu_frequency}")
```

Comment:
Is there a reason to add additional messaging here vs the uncore freq sweep & core freq sweep?  Should this be added to all experiments?

Reply:
That works for me, but it might be worthwhile to also file a quick feature request for when we update/refactor the experiment infrastructure - A high level human readable experiment config summary should be relatively easy to generate for any set of experiments based upon the data we already have